### DISCUSSION TITLE
Bots get stuck on ladders when trying to interact with items that are slightly out of reach

### DISCUSSION THREAD

**Discussion by NotWendy** [FakeFishGames/Barotrauma#13598] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

In this example, if the bot approaches the reactor from above to power it on, they'll get stuck once they climb onto the ladder.
![image](https://github.com/FakeFishGames/Barotrauma/assets/47554589/b973f8ad-e982-41f1-87b1-33d428060e0a)

Likely happens because the bot knows the interaction area of the reactor would be within reach under normal conditions, but does not realize that being on the ladder reduces their reach by a lot.
Normally, you can interact with an object from its edge and it'll drag your character towards the intended interaction point, which in this case is the panel at the bottom of the reactor. You can't get dragged this way while you're on the ladder, which is what confuses the bot.

Dragging the bot down to the floor manually (by jumping past them and grabbing them with G repeatedly) leads to them successfully powering the reactor on, as soon as they've gotten close to the bottom of the reactor.

Here is a submarine in which this behavior can be reproduced reliably. I've moved the engineer and the mechanic spawn points above the reactor, so they always get stuck on the ladder when you start the sub in the submarine editor.
[Gale.zip](https://github.com/FakeFishGames/Barotrauma/files/14638996/Gale.zip)

The issue has been reported by numerous people outside of GitHub, and most commonly happens with large objects.

### Reproduction steps

1. Download and unpack the example submarine.
2. Start it in the submarine editor's test mode.
3. Go to the reactor room to see two bots stuck on the ladder.

Alternatively, set up a similar structure yourself, forcing bots to approach a large interactable object from a ladder.

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.2.8.0 (Winter Update hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Thank you for the report! The issue has been addressed in our private repo now, fix coming up in a future unstable build.
